sbi/neural_nets/net_builders/vector_field_nets.py

```diff
@@ -0,0 +1,1186 @@
+
+        return estimator_cls(
+            net=vectorfield_net,
+            input_shape=batch_x[0].shape,
+            condition_shape=batch_y[0].shape,
+            embedding_net=embedding_net_y,
+            mean_0=mean_0,
+            std_0=std_0,
+        )
+    else:
+        raise ValueError(f"Unknown estimator type: {estimator_type}")
+
+
+# For backward compatibility
+def build_flow_matching_estimator(*args, **kwargs):
+    return build_vector_field_estimator(*args, estimator_type="flow", **kwargs)
+
+
+def build_score_matching_estimator(*args, **kwargs):
+    return build_vector_field_estimator(*args, estimator_type="score", **kwargs)
+
+
+# ======= Time Embedding Shared Components =======
+
+
+class SinusoidalTimeEmbedding(nn.Module):
+    """Sinusoidal time embedding as used in Vaswani et al. (2017).
+
+    Can be used for time embedding in both vector field nets.
+    """
+
+    def __init__(self, embed_dim: int = 16, max_freq: float = 0.01):
+        """Initialize sinusoidal embedding.
+
+        args:
+            embed_dim: dimension of the embedding (must be even)
+            max_freq: maximum frequency denominator
+        """
+        super().__init__()
+        if embed_dim % 2 != 0:
+            raise ValueError("embedding dimension must be even")
+
+        self.embed_dim = embed_dim
+        self.max_freq = max_freq
+        # compute frequency bands
+        div_term = torch.exp(
+            torch.arange(0, embed_dim, 2) * (-math.log(max_freq) / embed_dim)
+        )
+        self.register_buffer("div_term", div_term)
+        self.out_features = embed_dim
+
+    def forward(self, t: Tensor) -> Tensor:
+        """Embed time using transformer sinusoidal embeddings.
+
+        args:
+            t: time tensor of shape (batch_size, 1) or (batch_size,) or scalar ()
+
+        returns:
+            time embedding of shape (batch_size, embed_dim) or (embed_dim,)
+            for scalar input
+        """
+        # handle scalar inputs (0-dim tensors)
+        if t.ndim == 0:
+            # create output for a single time point
+            time_embedding = torch.zeros(self.embed_dim, device=t.device)
+            time_embedding[0::2] = torch.sin(t * self.div_term)
+            time_embedding[1::2] = torch.cos(t * self.div_term)
+            return time_embedding.unsqueeze(0)
+
+        # ensure time has the right shape for broadcasting
+        if t.ndim == 1:
+            t = t.unsqueeze(-1)
+
+        # create embeddings pe(pos, 2i) = sin(pos/1000^(2i/d))
+        # pe(pos, 2i+1) = cos(pos/1000^(2i/d))
+        time_embedding = torch.zeros(t.shape[:-1] + (self.embed_dim,), device=t.device)
+        time_embedding[:, 0::2] = torch.sin(t * self.div_term)
+        time_embedding[:, 1::2] = torch.cos(t * self.div_term)
+
+        return time_embedding
+
+
+class RandomFourierTimeEmbedding(nn.Module):
+    """Gaussian random features for encoding time steps.
+
+    This is to be used as a utility for score-matching."""
```

Comment:
I don't understand why sinusoidal embedding can be used for both vector field net types (Score/flow matching), but RandomFourierTimeEmbedding only for score-matching

Reply:
Yeah should both be compatible.